pkg/aws/aws.go

```diff
@@ -2,51 +2,60 @@ package aws
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/dynamodb"
-	"github.com/aws/aws-sdk-go/service/cloudwatch"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/pricing"
 
-	dynamoWaste "github.com/timmyers/cloudwaste/pkg/aws/dynamodb"
 	ec2Waste "github.com/timmyers/cloudwaste/pkg/aws/ec2"
-	"github.com/timmyers/cloudwaste/pkg/aws/util"
 )
 
+// func AnalyzeWaste() {
+// 	sess := session.Must(session.NewSession())
+// 	region := "us-east-1"
+// 	pricingClient := &ec2Waste.Pricing{
+// 		Client: pricing.New(sess, aws.NewConfig().WithRegion(region)),
+// 	}
+
+// 	pricingClient.GetUnusedEBSVolumePrice(context.TODO(), region)
+// }
+
 func AnalyzeWaste() {
 	sess := session.Must(session.NewSession())
 
-	client := &ec2Waste.EC2{
-		Client: ec2.New(sess, aws.NewConfig().WithRegion("us-east-1")),
-	}
-	dynamoClient := &dynamoWaste.DynamoDB{
-		DynamoDBClient: dynamodb.New(sess, aws.NewConfig().WithRegion("us-east-1")),
-		CloudwatchClient: cloudwatch.New(sess, aws.NewConfig().WithRegion("us-east-1")),
-	}
-
-	var unusedResources []util.AWSResourceObject
+	region := "us-east-1"
 
-	// Run all the checks
-	unusedAddresses, err := client.GetUnusedElasticIPAddresses(context.TODO())
-	if err == nil {
-		unusedResources = append(unusedResources, unusedAddresses...)
-	}
-	unusedGateways, err := client.GetUnusedNATGateways(context.TODO())
-	if err == nil {
-		unusedResources = append(unusedResources, unusedGateways...)
-	}
-	unusedVolumes, err := client.GetUnusedEBSVolumes(context.TODO())
-	if err == nil {
-		unusedResources = append(unusedResources, unusedVolumes...)
-	}
-	unusedDynamoDBTables, err := dynamoClient.GetUnusedDynamoDBTables((context.TODO()))
-	if err == nil {
-		unusedResources = append(unusedResources, unusedDynamoDBTables...)
+	client := &ec2Waste.Client{
+		EC2:     ec2.New(sess, aws.NewConfig().WithRegion(region)),
+		Pricing: pricing.New(sess, aws.NewConfig().WithRegion(region)),
 	}
 
-	for _, r := range unusedResources {
-		fmt.Printf("%s - %s\n", r.R.Type(), r.R.ID())
-	}
+	client.AnalyzeEBSVolumeWaste(context.TODO(), region)
```

Comment:
**[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Error return value of `client.AnalyzeEBSVolumeWaste` is not checked (errcheck)